src/models/UserProfileModel.ts

```diff
@@ -90,16 +95,29 @@ export class UserProfileModel extends Model {
      * @param userId The identifier of a user.
      */
     public async getActions(userId: string): Promise<UserAction[]> {
-        const actions = <UserAction[]>this.get(userId);
-        return (Array.isArray(actions) && actions) || this.fetchActions(userId);
+        let actions = <UserAction[]>this.get(userId);
+        actions = Array.isArray(actions) ? actions : await this.fetchActions(userId);
+
+        this.set(userId, actions, UserProfileModel.MODEL_CONFIG);
+
+        return actions;
+    }
+
+    /**
+     * Delete all actions related to a user from the model.
+     * 
+     * @param userId The identifier of a user.
+     */
+    public deleteActions(userId: string) {
+        this.set(userId, undefined, UserProfileModel.MODEL_CONFIG);
+        this.getOrFetchCache[userId] = undefined;
```

Comment:
So to delete the actions here are we setting it as `undefined`?

Reply:
Yess exactly 😄